app/assets/javascripts/admin/addon/components/schema-theme-setting/editor.gjs

```diff
@@ -47,20 +49,21 @@ export default class SchemaThemeSettingEditor extends Component {
   get tree() {
     let schema = this.schema;
     let data = this.data;
+    let tree = new Tree();
 
     for (const point of this.history) {
+      tree.propertyName = point.propertyName;
       data = data[point.node.index][point.propertyName];
       schema = schema.properties[point.propertyName].schema;
     }
 
-    const tree = new Tree();
-
     data.forEach((object, index) => {
       const node = new Node({
         index,
         schema,
         object,
-        text: object[schema.identifier],
+        text: object[schema.identifier] || `${schema.name} ${index + 1}`,
```

Comment:
Can we add a test for the default/fallback name when there's no identifier?

Reply:
Technically this has been tested because my system test is clicking on the link via the fallback name.

Reply:
But anyway, I can add a test for this here in the component test.